libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.spec.ts

```diff
@@ -231,6 +233,93 @@ describe('ModalWrapperComponent', () => {
+        const ionContentElement = spectator.query('ion-content');
+        const ionToolbarElement = spectator.query('ion-toolbar');
+        const embeddedComponentElement = ionContentElement.firstElementChild;
+        const embeddedPageProgress = embeddedComponentElement.querySelector('kirby-page-progress');
+        const pageProgressAsIonToolbarChild = ionToolbarElement.querySelector(
+          ':scope > kirby-page-progress'
+        );
+
+        expect(embeddedPageProgress).toBeNull();
+        expect(pageProgressAsIonToolbarChild).not.toBeNull();
+      });
+    });
+
+    describe('with dynamic page progress', () => {
+      beforeEach(() => {
+        spectator = modalWrapperTestBuilder
+          .flavor('modal')
+          .withDynamicPageProgress()
+          .build();
+        spectator.detectComponentChanges();
+      });
+
+      afterEach(() => {
+        // Ensure any observers are destroyed:
+        spectator.fixture.destroy();
+      });
+
+      it('should move embedded page progress to wrapper component when rendered', async () => {
+        const pageProgressContent = spectator.element.querySelector('kirby-page-progress');
+        expect(pageProgressContent).toBeNull();
+
+        const embeddedComponent = spectator.query(DynamicPageProgressEmbeddedComponent);
+        embeddedComponent.showPageProgress = true;
+        spectator.detectChanges();
+        await TestHelper.waitForResizeObserver();
+
+        const ionContentElement = spectator.query('ion-content');
+        const ionToolbarElement = spectator.query('ion-toolbar');
+        const embeddedComponentElement = ionContentElement.firstElementChild;
+        const embeddedPageProgress = embeddedComponentElement.querySelector('kirby-page-progress');
+        const pageProgressAsIonToolbarChild = ionToolbarElement.querySelector(
+          ':scope > kirby-page-progress'
+        );
+        expect(embeddedPageProgress).toBeNull();
+        expect(pageProgressAsIonToolbarChild).not.toBeNull();
+      });
+
+      it('should remove embedded page progress content from wrapper component when not rendered', async () => {
+        let pageProgress = spectator.element.querySelector('kirby-page-progress');
+        expect(pageProgress).toBeNull();
+
+        const embeddedComponent = spectator.query(DynamicPageProgressEmbeddedComponent);
+        embeddedComponent.showPageProgress = true;
+        spectator.detectChanges();
+        await TestHelper.waitForResizeObserver();
+
+        const ionToolbarElement = spectator.query('ion-toolbar');
+        const pageProgressAsIonToolbarChild = ionToolbarElement.querySelector(
+          'kirby-page-progress'
+        );
+        expect(pageProgressAsIonToolbarChild).not.toBeNull();
+
+        embeddedComponent.showPageProgress = false;
+        spectator.detectChanges();
+
+        pageProgress = pageProgressAsIonToolbarChild.querySelector(':scope > kirby-page-progress');
+        expect(pageProgress).toBeNull();
```

Comment:
This looks for a kirby-page-progress as a child of  kirby-page-progress... ;)
```suggestion
        pageProgressAsIonToolbarChild = ionToolbarElement.querySelector(':scope > kirby-page-progress');
        expect(pageProgressAsIonToolbarChild).toBeNull();
```

Reply:
Yes... Yes it does...